src/detect-parse.c

```diff
@@ -604,6 +604,26 @@ static int SigParseOptions(DetectEngineCtx *de_ctx, Signature *s, char *optstr,
         }
     }
 
+    /* Validate double quoting, trimming trailing white space along the way. */
+    if (strlen(optvalue) > 0) {
+        size_t ovlen = strlen(optvalue);
+        if (optvalue[0] == '"') {
+            for (; ovlen > 0; ovlen--) {
+                if (isblank(optvalue[ovlen - 1])) {
+                    optvalue[ovlen - 1] = '\0';
+                } else {
+                    break;
+                }
+            }
+            if (optvalue[ovlen - 1] != '"') {
```

Comment:
<pre>
detect-parse.c: In function ‘SigParse’:
detect-parse.c:624:25: error: array subscript is above array bounds [-Werror=array-bounds]
             if (optvalue[ovlen - 1] != '"') {
                         ^
</pre>